rust/rsmgp-example/src/lib.rs

```diff
@@ -1,116 +1,131 @@
-use backtrace::Backtrace;
 use c_str_macro::c_str;
 use rsmgp_sys::memgraph::*;
 use rsmgp_sys::mgp::*;
 use rsmgp_sys::property::*;
 use rsmgp_sys::result::*;
 use rsmgp_sys::rsmgp::*;
 use rsmgp_sys::value::*;
-use rsmgp_sys::{close_module, define_procedure, define_type, init_module};
+use rsmgp_sys::{close_module, define_optional_type, define_procedure, define_type, init_module};
 use std::ffi::CString;
 use std::os::raw::c_int;
 use std::panic;
 
-init_module!(|memgraph: &Memgraph| -> Result<(), MgpError> {
+init_module!(|memgraph: &Memgraph| -> MgpResult<()> {
     memgraph.add_read_procedure(
         test_procedure,
         c_str!("test_procedure"),
+        &[],
+        &[],
         &[
-            define_type!("labels_count", FieldType::Int),
-            define_type!("has_L3_label", FieldType::Bool),
-            define_type!("first_label", FieldType::String),
-            define_type!("name_property", FieldType::String),
-            define_type!("properties_string", FieldType::String),
-            define_type!("first_edge_type", FieldType::String),
-            define_type!("list", FieldType::List, FieldType::Int),
+            define_type!("labels_count", Type::Int),
+            define_type!("has_L3_label", Type::Bool),
+            define_type!("first_label", Type::String),
+            define_type!("name_property", Type::String),
+            define_type!("properties_string", Type::String),
+            define_type!("first_edge_type", Type::String),
+            define_type!("list", Type::List, Type::Int),
         ],
     )?;
 
     memgraph.add_read_procedure(
         basic,
         c_str!("basic"),
-        &[define_type!("input_string", FieldType::String)],
+        &[define_type!("input_string", Type::String)],
+        &[define_optional_type!(
+            "optional_input_int",
+            &MgpValue::make_int(0, &memgraph)?,
+            Type::Int
+        )],
+        &[
+            define_type!("output_string", Type::String),
+            define_type!("output_int", Type::Int),
+        ],
     )?;
+
     Ok(())
 });
 
-define_procedure!(basic, |memgraph: &Memgraph| -> Result<(), MgpError> {
+define_procedure!(basic, |memgraph: &Memgraph| -> MgpResult<()> {
     let result = memgraph.result_record()?;
-    result.insert_string(c_str!("input_string"), c_str!("TODO: Add input param."))?;
+    let args = memgraph.args()?;
+    let output_string = args.value_at(0)?;
+    let output_int = args.value_at(1)?;
```

Comment:
This is a bit confusing: I would rename them to `input_....` (to show that this is how input parameters can be read) and I would add a small comment about this is just forwards the input parameters as outputs.